submissions/SamVal007/Friends_APP/script.js

```diff
@@ -0,0 +1,100 @@
+const CONTAINER = document.getElementById('container');
+const SIDEBAR = document.getElementById('sidebar');
+const SEARCH_FIELD = document.getElementById('searchField');
+const URL = 'https://randomuser.me/api/?results=40';
+
+let persons = [];
+let displayList = [];
+
+function getPeopleData() {
+  let response = fetch(URL)
+    .then(
+      successResponse => {
+        if (successResponse.status != 200) {
+          return null;
+        } else {
+          return successResponse;
+        }
+      }
+    )
+    .then(data => data.json())
+    .then(function (data) {
+      persons = data.results;
+      displayList = persons;
+      displayCards();
+    })
+    .catch(function (error) {
+      console.log('We have an error: ', error);
+      alert('Please, check your internet connection and try again');
+    })
+}
+
+function getTemplate(person) {
+  return `<div class="card">
+            <div class="picture">
+              <img src="${person.picture.large}">
+            </div>
+            <h1>${person.name.last} ${person.name.first}</h1>
+            <div class="cardInfo">
+              <p><span class="text_title">Age:</span> ${person.dob.age}</p>
+              <p><span class="text_title">Gender:</span> ${person.gender}</p>
+              <p><span class="text_title">Phone:</span> ${person.phone}</p>
+              <p><span class="text_title">Country:</span> ${person.location.country}</p>
+              <p><span class="text_title">City:</span> ${person.location.city}</p>
+            </div>
+          </div>`
+}
+
+function displayCards() {
+  CONTAINER.innerHTML = '';
+  let cardMarkup = '';
+  cardMarkup += displayList.reduce((accumulator, currentValue) => accumulator.concat(getTemplate(currentValue)), '');
+  CONTAINER.insertAdjacentHTML('afterbegin', cardMarkup);
+}
+
+document.addEventListener('DOMContentLoaded', function () {
+  getPeopleData();
+
+  SIDEBAR.addEventListener('click', function (event) {
+    let radio = event.target.closest('.rdBtn');
+    let btn = event.target.closest('button');
+
+    if (radio) {
+
+      switch (radio.id) {
+        case 'genderAll':
+          displayList = persons;
+          break;
+        case 'genderMale':
+          displayList = persons.filter(el => el.gender === 'male');
+          break;
+        case 'genderFemale':
+          displayList = persons.filter(el => el.gender === 'female');
+          break;
+        case 'nameAsc':
+          displayList = displayList.sort((b, a) => a.name.last > b.name.last ? -1 : 1);
+          break;
+        case 'nameDesc':
+          displayList = displayList.sort((b, a) => a.name.last < b.name.last ? -1 : 1);
+          break;
+        case 'ageLow':
+          displayList = displayList.sort((a, b) => a.dob.age - b.dob.age);
+          break;
+        case 'ageHigh':
+          displayList = displayList.sort((b, a) => a.dob.age - b.dob.age);
```

Comment:
1. Why you need assigning to `displayList`?
2. Same function as at line 81. DRY

Reply:
Also, please check this comment https://github.com/kottans/frontend-2021-homeworks/pull/449#discussion_r578247714

Reply:
Thank you!
Yes, really, i dont need this 'assigning '!
Ok, I will read this comments.

Reply:
I'm sorry, Did I understand correctly -  I should  remake this part of code (this switch is too long and it's not "Single responsibility"):

  if (radio) {
      switch (radio.id) {
        case 'genderAll':
          displayList = persons;
          break;
        case 'genderMale':
          displayList = persons.filter(el => el.gender === 'male');
          break;
        case 'genderFemale':
          displayList = persons.filter(el => el.gender === 'female');
          break;
        case 'nameAsc':
          displayList.sort((b, a) => a.name.last > b.name.last ? -1 : 1);
          break;
        case 'nameDesc':
          displayList.sort((b, a) => a.name.last < b.name.last ? -1 : 1);
          break;
        case 'ageLow':
          displayList.sort((a, b) => a.dob.age - b.dob.age);
          break;
        case 'ageHigh':
          displayList.sort((b, a) => a.dob.age - b.dob.age);
          break;
      }

With adding functions for sorting.
BTW, Should I delete "long switch" and make few different "addEventListneres"   for each case of switch or it will be very ig load on site? (sorry may be for stupid questions, but i just want to understand)

Reply:
1. Yes, you should delete the long switch
2. You can have a listener on the form and listen to `change` event. Or you can separate listeners between fieldsets. Up to you

Reply:
Huh! Unbelievable, but seems I did it!:)
I dodnt know why, but this refactoring took me quit long time. 
Basically, I think. It because I start fixing everything at one time... it was big mistake -  nothing work and could understand why. So, then I started do it step-by-step, like You teaching us:) Refactored sort logic, then long switch. 
Sorry, may be It's not necessary to write it here, but whom else can i write it;))

Ok, so now it's seems